Image Feed/ImagesList/ImagesListViewController.swift

```diff
@@ -1,73 +1,138 @@
+                print(error)
+            }
+        }
+        
+        let date = photos[indexPath.row].createdAt
+        cell.dateLabel.text = dateFormatter.string(from: date)
+        let isLiked = photos[indexPath.row].isLiked
+        let buttonImage = isLiked ? UIImage(named: "like_button_on") : UIImage(named: "like_button_off")
+        cell.likeButton.setImage(buttonImage, for: .normal)
+    }
+    
+    
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: ImagesListCell.reuseIdentifier, for: indexPath)
         
+        let cell = tableView.dequeueReusableCell(withIdentifier: ImagesListCell.reuseIdentifier, for: indexPath)
         guard let imageListCell = cell as? ImagesListCell else {
             return UITableViewCell()
         }
-        
+        imageListCell.delegate = self
         configCell(for: imageListCell, with: indexPath)
-        
         return imageListCell
     }
-}
-
-extension ImagesListViewController {
-    func configCell(for cell: ImagesListCell, with indexPath: IndexPath) {
-        guard let image = UIImage(named: photosName[indexPath.row]) else {
-            return
-        }
-        
-        cell.cellImage.image = image
-        cell.dateLabel.text = dateFormatter.string(from: Date())
-        
-        let isLiked = indexPath.row % 2 == 0
-        let likeImage = isLiked ? UIImage(named: "like_button_on") : UIImage(named: "like_button_off")
-        cell.likeButton.setImage(likeImage, for: .normal)
+    
+    func updateTableViewAnimated() {
+        let oldCount = photos.count
+        let newCount = imagesListService.photos.count
+        photos = imagesListService.photos
+        guard oldCount != newCount else {return}
+        tableView.performBatchUpdates {
+            let indexPaths = (oldCount..<newCount).map { i in
+                IndexPath(row: i, section: 0)
+            }
+            tableView.insertRows(at: indexPaths, with: .automatic)
+        } completion: { _ in }
     }
 }
 
+extension ImagesListViewController: UITableViewDataSource{}
+
 extension ImagesListViewController: UITableViewDelegate {
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        performSegue(withIdentifier: showSingleImageSegueIdentifier, sender: indexPath)
+        performSegue(withIdentifier: ShowSingleImageSegueIdentifier, sender: indexPath)
     }
 }
+
+extension ImagesListViewController: ImagesListCellDelegate {
+    
+    func imageListCellDidTapLike(_ cell: ImagesListCell) {
+        guard let indexPath = tableView.indexPath(for: cell) else { return }
+        let photo = photos[indexPath.row]
+        guard let token = tokenStorage.token else {return}
+        UIBlockingProgressHUD.show()
+        imagesListService.changeLike(photoId: photo.id, isLike: !photo.isLiked, token: token, cell.setIsLiked)
+        UIBlockingProgressHUD.dismiss()
```

Comment:
давай поправим аргумент комплишена на (index: Int, isLiked: Bool) -> Void, и реализуем его здесь 
```suggestion
        imagesListService.changeLike(photoId: photo.id, isLike: !photo.isLiked, token: token) { [weak self] index, isLiked in 
         if let cell  = tableView. tableView.cellForRow(at: IndexPath(row: index, section: 0) as? ImagelistCell {
            cell.setLiked(isLiked)
          UIBlockingProgressHUD.dismiss()
         }
         
        }
        UIBlockingProgressHUD.dismiss()
```